### ISSUE TITLE
forge.op.loss.CrossEntropyLoss accuracy regression

### ISSUE THREAD

**Issue by philei-tt** [tenstorrent/tt-metal#27908]
Forge team noticed a small regression in forge.op.loss.CrossEntropyLoss after [Replace truncating fp32->bf16 with tie-to-even rounding rule](https://github.com/tenstorrent/tt-metal/commit/c724045b23). 

Here's how to reproduce it: 

```python
import ttnn
import torch

def cross_entropy_loss_ttnn(predictions, labels, reduction_constant):
    reduction_const_reshaped = ttnn.reshape(reduction_constant, (1, 1))
    softmax_predictions = ttnn.softmax(predictions, dim=-1)
    log_softmax = ttnn.log(softmax_predictions)
    weighted_log_probs = ttnn.multiply(labels, log_softmax)
    sum_per_sample = ttnn.sum(weighted_log_probs, dim=-1, keepdim=True)
    scaled_loss = ttnn.multiply(sum_per_sample, reduction_const_reshaped)
    final_loss = ttnn.mean(scaled_loss, dim=0, keepdim=True)
    return final_loss

def main():
    device = ttnn.open_device(device_id=0)
    
    try:
        torch.manual_seed(0)
        
        predictions = torch.randn(3, 5)
        target_indices = torch.empty(3, dtype=torch.long).random_(5)
        target_onehot = torch.nn.functional.one_hot(target_indices, num_classes=5).float()
        
        pytorch_result = torch.nn.CrossEntropyLoss()(predictions, target_indices)
        print(f"PyTorch: {pytorch_result.item():.6f}")
        
        predictions_ttnn = ttnn.from_torch(predictions, dtype=ttnn.float32, layout=ttnn.ROW_MAJOR_LAYOUT, device=device)
        labels_ttnn = ttnn.from_torch(target_onehot, dtype=ttnn.float32, layout=ttnn.ROW_MAJOR_LAYOUT, device=device)
        reduction_const_ttnn = ttnn.from_torch(torch.tensor([-1.0]), dtype=ttnn.float32, device=device)
        
        ttnn_result = ttnn.to_torch(cross_entropy_loss_ttnn(predictions_ttnn, labels_ttnn, reduction_const_ttnn))
        print(f"TTNN: {ttnn_result.item():.6f}")
        print(f"Difference: {abs(pytorch_result.item() - ttnn_result.item()):.6f}, Percentage: {abs(pytorch_result.item() - ttnn_result.item()) / pytorch_result.item() * 100:.6f}%")
        
    finally:
        ttnn.close_device(device)

if __name__ == "__main__":
    main()
```

Currently this script gives the following output: 
```
PyTorch: 2.272947
TTNN: 2.306580
Difference: 0.033633, Percentage: 1.479688%
```

And with truncation instead of rounding: 
```
PyTorch: 2.272947
TTNN: 2.293091
Difference: 0.020144, Percentage: 0.886239%
```
